src/main/java/com/uber/jenkins/phabricator/CommentBuilder.java

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.uber.jenkins.phabricator;
+
+import com.uber.jenkins.phabricator.utils.CommonUtils;
+import com.uber.jenkins.phabricator.utils.Logger;
+import hudson.model.Result;
+import hudson.plugins.cobertura.Ratio;
+import hudson.plugins.cobertura.targets.CoverageMetric;
+import hudson.plugins.cobertura.targets.CoverageResult;
+
+public class CommentBuilder {
+    private static final String UBERALLS_TAG = "uberalls";
+    private final Logger logger;
+    private final CoverageResult currentCoverage;
+    private final StringBuilder comment;
+    private final String buildURL;
+    private final Result result;
+
+    public CommentBuilder(Logger logger, Result result, CoverageResult currentCoverage, String buildURL) {
+        this.logger = logger;
+        this.result = result;
+        this.currentCoverage = currentCoverage;
+        this.buildURL = buildURL;
+        this.comment = new StringBuilder();
+    }
+
+    /**
+     * Get the final comment to post to Phabricator
+     * @return
+     */
+    public String getComment() {
+        return comment.toString();
+    }
+
+    /**
+     * Determine whether to attempt to process coverage
+     * @return
+     */
+    public boolean hasCoverageAvailable() {
+        if (currentCoverage == null) {
```

Comment:
Can simplify this to:

return currentCoverage != null && currentCoverage.getCoverage(CoverageMetric.LINE) != null;